docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+interface Product {
+  id: ID!
+  sku: String
+  package: String
+  createdBy: User
+  hidden: String @inaccessible
+}
+```
+
+
+### `@key`
+
+```graphql
+directive @key(fields: FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
+```
+
+The `@key` directive indicates a combination of fields that can be used to uniquely identify a particular instance of a federated entity:
+
+```graphql
+type Product @key(fields: "upc") {
+  upc: ID!
+  name: String
+}
+```
+
+Multiple keys can be defined on a single object type:
+
+```graphql
+type Product @key(fields: "upc") @key(fields: "sku") {
+  upc: ID!
+  sku: ID!
+  name: String
+}
+```
+
+> **Note:** Repeated directives (such as `@key` in the example above) require support by your underlying GraphQL implementation. Some subgraph libraries support only a single `@key` per entity because of this limitation.
+
+### `@link`
+
+```graphql
+directive @link(
+  url: String,
+  as: String,
+  for: link__Purpose,
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive links definitions within the document to external schemas.
+
+External schemas are identified by their `url`, which optionally ends with a name and version with the following format: `{NAME}`/v`{MAJOR}`.`{MINOR}`
+
+The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).
+
+The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas that contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.
+
+By default, `@link`ed definitions will be namespaced, i.e., `@federation__requires`. The `as` argument lets you pick the name for this namespace:
+```graphql
+extend schema @link(url: "https://specs.apollo.dev/federation/v2.0", as: "fed2")
+type User {
+  metrics: Metrics @external
+  favorites: UserFavorites @fed2__requires(fields: "metrics")
+}
+```
+
+The `import` argument enables you to import external definitions into your namespace, so they are not prefixed.
+
+```graphql
+    schema
+      @link(url: "https://specs.apollo.dev/link/v1.0")
+      @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", { name: "@tag", as: "@mytag" }, "@extends", "@shareable", "@inaccessible", "@override"])
+```
+
+In the example above, we import various directives from `federation/v2.0` into our namespace. We also rename one of them, bringing in federation's `@tag` as `@mytag` to distinguish it from a different `@tag` directive already in the schema.
+
+### `@override`
+
+```graphql
+directive @override(from: String!) on FIELD_DEFINITION
+```
+
+The `@override` directive indicates that the current subgraph takes responsibility for resolving a field _away_ from the subgraph specified in the `from` argument.
```

Comment:
do we need to call out that the subgraph name has to match the one in the studio?